### ISSUE TITLE
microcontroller ops

### ISSUE THREAD

**Issue by wendy518** [tensorflow/tensorflow#31035]
HI, i just want to know in all_ops_resolver.cc 
it only has these ops for microcontroller.

> TfLiteRegistration* Register_DEPTHWISE_CONV_2D();
> TfLiteRegistration* Register_FULLY_CONNECTED();
> TfLiteRegistration* Register_SOFTMAX();
> TfLiteRegistration* Register_CONV_2D();
> TfLiteRegistration* Register_AVERAGE_POOL_2D();
> TfLiteRegistration* Register_MAX_POOL_2D();
> TfLiteRegistration* Register_ABS();
> TfLiteRegistration* Register_PRELU();
>

[https://github.com/tensorflow/tensorflow/blob/master/tensorflow/lite/experimental/micro/kernels/all_ops_resolver.cc](url)

And in hello world test, the keras frame can be used, so i want to know if it is possible to use the model with tf.keras.layers.Conv2D and tf.keras.layers.MaxPool2D?

And if is possible to use tf.keras.layers.Convolution1D and tf.keras.layers.MaxPooling1D?

Besides, if the model can be converted to .tflite means that the model can be deploy to the sparkfun board?

thank you

**Comment by gadagashwini** [comment]
Please provide details about what platform you are using (operating system, architecture). Also include your TensorFlow version. Also, did you compile from source or install a binary?

Make sure you also include the exact command if possible to produce the output included in your test case. If you are unclear what to include see the issue template displayed in [the Github new issue template](https://github.com/tensorflow/tensorflow/issues/new/choose).

We ask for this in the issue submission template, because it is really difficult to help without that information. Thanks!

**Comment by WendyWangxyue** [comment]
I use the Linux Ubuntu 16.04 in VMware with Tensorflow 1.14 with CPU. I install Tensorflow with pip.
Build label: 0.27.1
Python 3.5.2
[GCC 5.4.0 20160609] on linux
tf.version
'1.14.0'

**Comment by gadagashwini** [comment]
@wendy518 Could you please elaborate issue with context. Will it be possible to provide the minimal reproducible code. Thanks!

**Comment by wendy518** [comment]
`model = tf.keras.models.Sequential()
model.add(tf.keras.layers.Convolution2D(30, (1,1), activation='relu',input_shape = (X_train.shape[1], X_train.shape[2],1)))
model.add(tf.keras.layers.MaxPooling2D(pool_size=(2,2), padding='same'))
model.add(tf.keras.layers.Flatten())
model.add(tf.keras.layers.Dense(units =20,kernel_initializer = 'uniform', activation = 'relu'))
model.add(tf.keras.layers.Dense(units = 5,kernel_initializer = 'uniform', activation = 'softmax'))`

I want to know if this model can be used in the microcontroller, since the ops file does not represent clearly, if the high API can be used.
Thank you

**Comment by njeffrie** [comment]
I think we support all the ops you need for the model you posted.  The best way to evaluate this is to train your model, convert to a micro-compatible TFLite model using the [TFLite converter](https://www.tensorflow.org/lite/convert/python_api), then follow the instructions [here](https://www.tensorflow.org/lite/microcontrollers/build_convert) to run the model on a microcontroller platform like Sparkfun Edge.  If you run into any bugs along the way, please file a bug and we will look into it.
